pkg/store/connection/connection_recorder.go

```diff
@@ -161,3 +199,38 @@ func computeHash(bytes []byte) (string, error) {
 
 	return fmt.Sprintf("%x", hash), nil
 }
+
+func removeConnectionsForStates(c *Recorder, connectionID string) error {
+	itr := c.transientStore.Iterator(getConnectionStateKeyPrefix()(
+		connectionID),
+		getConnectionStateKeyPrefix()(connectionID)+"~",
+	)
+	defer itr.Release()
+
+	for itr.Next() {
+		key := string(itr.Key())
+		err := c.transientStore.Delete(key)
+
+		if err != nil {
+			return fmt.Errorf(
```

Comment:
Should not we check an error here? e.g
```
for itr.Next() {
 	if itr.Error() != nil {
 	 	return nil, itr.Error()
 	}
```

Reply:
True, but it should be placed before itr.Next() sequence, and so I did.

Reply:
`Release()` should be placed after the loop. An error might occur during the iteration.
e.g 
```
for itr.Next() {
	...
}

return iter.Error()
```

Reply:
Yes, you are right, I will place `itr.Error()` check after the loop. But `Release()` already has `defer`, so is it needed to be put after the loop?

Reply:
I am not sure what kind of error it might return. But the right place for it will be after the loop.

Reply:
But it is one `return` inside the loop, so if `Release()` will be put after the loop we can it will not be called in case of we get error in the loop. But we need to call it after we done with iterator. That's how it is now:
```
func removeConnectionsForStates(c *Recorder, connectionID string) error {
	itr := c.transientStore.Iterator(getConnectionStateKeyPrefix()(
		connectionID),
		getConnectionStateKeyPrefix()(connectionID)+"~",
	)
	defer itr.Release() // Release() is placed before any return so in case of any outcome it will be called

	for itr.Next() {
		key := string(itr.Key())
		err := c.transientStore.Delete(key)

		if err != nil {
			return fmt.Errorf( // this return can be missed if Release() would be placed after the loop
				"unable to delete connection state record from the transient store: key=%s connectionid=%s err=%w",
				key, connectionID, err)
		}
	}

	return itr.Error()
}
```

Reply:
As described here https://gobyexample.com/defer, deferred method will be executed at the end of enclosing function. So it is much safer to put it at the beginning of the function before any of the possible ends of this function (`return`s), than to put it after such point located inside the loop. In first case `itr.Release()` will be executed in any possibile end of the function, otherwise it will be executed only if we pass the loop.

Reply:
sorry for that typo)  `Release()` ->`Error()`
I meant `Error()` should be placed after the loop. An error might occur during the iteration.`

Reply:
Ahh, got it :D